src/commands/NewIntegrationFileCommand.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as kamel from './../kamel';
+
+const validFilename = require('valid-filename');
+
+const LANGUAGES = ['Java', 'xml', 'Yaml', 'Groovy', 'JavaScript', 'Kotlin'];
+
+export async function create() {
+	let language = await vscode.window.showQuickPick(LANGUAGES);
+	if (language) {
+		let workspaceFolder = await vscode.window.showWorkspaceFolderPick();
+		if (workspaceFolder) {
+			let filename = await vscode.window.showInputBox({
+				prompt: 'Please provide a name for the new file (without extension)',
+				validateInput: (name: string) => {
+					return validateFileName(name, language as string, workspaceFolder as vscode.WorkspaceFolder);
```

Comment:
You can omit `as string` if you define the const using a datatype definition. for instance `const language : string = await ....`.

Reply:
same applies to the folder